sklearn/ensemble/tests/test_partial_dependence.py

```diff
@@ -253,3 +260,32 @@ def test_plot_partial_dependence_multiclass():
     assert_raises(ValueError, plot_partial_dependence,
                   clf, iris.data, [0, 1],
                   grid_resolution=grid_resolution)
+
+
+def test_warning_raised_partial_dependence():
```

Comment:
We can factorize this 2 functions:

```python
@pytest.mark.parametrize(
    "func, params",
    [(partial_dependence, {'target_variables': [0], 'X': boston.data}),
     pytest.param(
         plot_partial_dependence,
         {'X': boston.data, 'features': [0, 1, (0, 1)]},
         marks=skip_if_no_matplotlib
     )]
)
def test_raise_deprecation_warning(func, params):
    clf = GradientBoostingRegressor(n_estimators=10, random_state=1)
    clf.fit(boston.data, boston.target)
    grid_resolution = 25

    warn_msg = "The function ensemble.{} has been deprecated".format(
        func.__name__
    )
    with pytest.warns(DeprecationWarning, match=warn_msg):
        func(clf, **params, grid_resolution=grid_resolution)
```

Reply:
Ok @glemaitre but honestly, this code  will be removed in 0.23, and untouched until then. It does not need to follow the best coding practices.